sunpy/map/sources/hinode.py

```diff
@@ -36,24 +46,43 @@ class XRTMap(GenericMap):
                                   "Be_thick", "Gband", "Ti_poly"]
 
     def __init__(self, data, header, **kwargs):
-        super().__init__(data, header, **kwargs)
-
-        fw1 = header.get('EC_FW1_')
+        fw1 = header.get('EC_FW1_', '')
         if fw1.lower() not in _lower_list(self.filter_wheel1_measurements):
-            raise ValueError('Unpexpected filter wheel 1 in header.')
-
-        fw2 = header.get('EC_FW2_')
+            raise ValueError(f'Unexpected filter wheel 1 {fw1} in header.')
+        fw2 = header.get('EC_FW2_', '')
         if fw2.lower() not in _lower_list(self.filter_wheel2_measurements):
-            raise ValueError('Unpexpected filter wheel 2 in header.')
-
+            raise ValueError(f'Unexpected filter wheel 2 {fw2} in header.')
+        super().__init__(data, header, **kwargs)
         self.plot_settings['cmap'] = 'hinodexrt'
 
     @property
     def _timesys(self):
         if self.meta.get('timesys', '').upper() == 'UTC (TBR)':
             return 'UTC'
-        else:
-            return super()._timesys
+        return super()._timesys
+
+    @property
+    # @cached_property_based_on('_meta_hash')
+    def observer_coordinate(self):
+        # Based on the discussion from
+        # # https://community.openastronomy.org/t/sunpymetadatawarnings-when-using-hinode-xrt-data/393/7
+        # Certain keywords are missing from the level 1 header that make haev to set values
+        # to avoid warnings being raised.
+        missing_meta = {}
+        for keys, kwargs in self._supported_observer_coordinates:
+            if not isinstance(kwargs['frame'], str):
+                kwargs['frame'] = kwargs['frame'].name
+            missing_meta[kwargs['frame']] = set(keys).difference(self.meta.keys())
+            if "heliographic_stonyhurst" in missing_meta:
+                if "hgln_obs" in missing_meta["heliographic_stonyhurst"]:
+                    extra_kwargs = {"lon": 0,
+                                    "lat": self.meta.get("SOLAR_B0", sun.B0(self.meta.get("DATE_OBS")).to_value(u.deg)),
+                                    "radius": constants.radius
+                                    }
+                    sc = SkyCoord(obstime=self.date, **{**kwargs, **extra_kwargs})
+                    sc = sc.heliographic_stonyhurst
+                    return SkyCoord(sc.replicate(rsun=self._rsun_meters(sc.radius)))
+        return super().observer_coordinate
```

Comment:
```suggestion
    @property
    def _supported_observer_coordinates(self):
        return (super()._supported_observer_coordinates
                + [(('solar_b0', 'dsun_obs'), {'lon': 0*u.deg,
                                               'lat': self.meta.get('solar_b0'),
                                               'radius': self.meta.get('dsun_obs'),
                                               'unit': (u.deg, u.deg, u.m),
                                               'frame': "heliographic_stonyhurst"})])
```
Note that unlike our other map sources, we add this option with the assumption at the end of the list so that we'll automatically use observer metadata if it already exists.